pkg/operator/ceph/cluster/cluster.go

```diff
@@ -71,19 +71,20 @@ type clusterHealth struct {
 }
 
 func newCluster(c *cephv1.CephCluster, context *clusterd.Context, csiMutex *sync.Mutex, ownerRef *metav1.OwnerReference) *cluster {
+	// at this phase of the cluster creation process, the identity components of the cluster are
+	// not yet established. we reserve this struct which is filled in as soon as the cluster's
+	// identity can be established.
+	clusterInfo := client.AdminClusterInfo(c.Namespace)
```

Comment:
I didn't really like this when I did this. I wonder if it is better not to set it and allow the operator to crash so we can catch the bugs where the operator is referring to an incomplete ClusterInfo. This still needs more thought...

Reply:
Maybe we can leave that for later cleanup?